tests/lib/rules/no-restricted-imports.js

```diff
@@ -1953,6 +2008,126 @@ ruleTester.run("no-restricted-imports", rule, {
             endColumn: 9,
             message: "* import is invalid because import name matching '/^Foo/u' pattern from 'foo' is restricted from being used."
         }]
+    },
+    {
+        code: "export { Bar } from 'foo';",
+        options: [{
+            patterns: [{
+                group: ["foo"],
+                allowImportNamePattern: "^Foo",
+                message: "Only imports that match the pattern '/^Foo/u' are allowed to be imported from 'foo'."
+            }]
+        }],
+        errors: [{
+            type: "ExportNamedDeclaration",
+            line: 1,
+            column: 10,
+            endColumn: 13,
+            message: "'Bar' import from 'foo' is restricted because only imports that match the pattern '/^Foo/u' are allowed from 'foo'. Only imports that match the pattern '/^Foo/u' are allowed to be imported from 'foo'."
+        }]
+    },
+    {
+        code: "import { AllowedObject, DisallowedObject } from \"foo\";",
+        options: [{
+            paths: [{
+                name: "foo",
+                allowImportNames: ["AllowedObject"],
+                message: "Only 'AllowedObject' is allowed to be imported from 'foo'."
+            }]
+        }],
+        errors: [{
+            message: "'DisallowedObject' import from 'foo' is restricted because only 'AllowedObject' import(s) is/are allowed. Only 'AllowedObject' is allowed to be imported from 'foo'.",
+            type: "ImportDeclaration",
+            line: 1,
+            column: 25,
+            endColumn: 41
+        }]
+    },
+    {
+        code: "import { AllowedObject, DisallowedObject } from \"foo\";",
+        options: [{
+            patterns: [{
+                group: ["foo"],
+                allowImportNames: ["AllowedObject"],
+                message: "Only 'AllowedObject' is allowed to be imported from 'foo'."
+            }]
+        }],
+        errors: [{
+            message: "'DisallowedObject' import from 'foo' is restricted because only 'AllowedObject' import(s) is/are allowed. Only 'AllowedObject' is allowed to be imported from 'foo'.",
+            type: "ImportDeclaration",
+            line: 1,
+            column: 25,
+            endColumn: 41
+        }]
+    },
+    {
+        code: "import { AllowedObject, DisallowedObject } from \"foo\";",
+        options: [{
+            patterns: [{
+                group: ["foo"],
+                allowImportNames: ["AllowedObject"],
+                allowImportNamePattern: "^Allow",
+                message: "Only 'AllowedObject' is allowed to be imported from 'foo'."
```

Comment:
```suggestion
                allowImportNames: ["AllowedObject"],
                message: "Only 'AllowedObject' is allowed to be imported from 'foo'."
```
this can be removed as per new validation

Reply:
We could remove the whole test case, as this case is already the previous case.

Reply:
Ya that can also be done as it is causing the failing of ci